python/pdo/contract/response.py

```diff
@@ -90,6 +100,216 @@ class ContractResponse(object) :
+                commit_dependencies = response.txn_params['dependency_list_commit_ids']
+                fail_explit_commit_dependencies = False
+                for commit_id_temp in commit_dependencies:
+                    txnid = transaction_dependencies.FindDependencyLocally(commit_id_temp[0], crypto.byte_array_to_base64(commit_id_temp[1]))
+                    if txnid :
+                        txn_dependencies.append(txnid)
+                    else:
+                        fail_explit_commit_dependencies = True
+                        break
+            
+                if fail_explit_commit_dependencies:
+                    return submitted_any
+                    
+                # OK, all commit dependencies are met. Add any transaction dependecies explicitly specified by client durind the commit call.
+                # these will be checked by the submitter
+                for txn_id in response.txn_params['dependency_list_txnids']:
+                    txn_dependencies.append(txn_id)                         
+                
+                # get rest of the info needed to submit txn
+                ledger_config = response.txn_params['ledger_config']
+                wait = response.txn_params['wait']
+            
+                # submit txn
+                try:
+                    txn_id =  response.submit_update_transaction(ledger_config, wait=wait, transaction_dependency_list=txn_dependencies)
+                    if txn_id:
+                        logger.info("Submitted transaction for request number %d", request_number)
+                        submitted_any = True
+                        del rep_completed_but_txn_not_submitted_updates[contract_id][request_number] # remove the task from the pending list
+                        # acquire lock to the condition varaiable, add the commit_id to completed list, notify, and release the lock
+                        cls.txn_completion_condition.acquire()
+                        cls.set_of_completed_txns.add(commit_id)
+                        cls.txn_completion_condition.notify()
+                        cls.txn_completion_condition.release()
+                    else:
+                        logger.error("Did not get a transaction id after transaction submission,  request nunmber %d", request_number)
+                        cls.transaction_submission_exceptions_queue.put("Did not get a transaction id after transaction submission")
+                        raise Exception("Did not get a transaction id after transaction submission,  request nunmber %d", request_number)
+                except Exception as e:
+                    logger.error("Transaction submission failed, request nunmber %d %s", request_number, str(e))
+                    cls.transaction_submission_exceptions_queue.put("Transaction submission failed")
+                    raise Exception("Transaction submission failed, request nunmber %d %s", request_number, str(e))
+            
```

Comment:
here's that pattern again with three different variants of the same reporting. In this case you should just drop the inner exception since it gets masked and repurposed by the out catch